test/helpers/kubectl.go

```diff
@@ -1784,14 +1789,23 @@ func (kub *Kubectl) CiliumStatusPreFlightCheck() error {
 	return nil
 }
 
-// CiliumControllersPreFlightCheck validates that all controllers are not
-// failing. If any of the controllers fails will return an error.
-func (kub *Kubectl) CiliumControllersPreFlightCheck() error {
-	ginkgoext.By("Checking that no controllers in Cilium have failed")
+// CiliumStatusPreFlightCheck validates the cilium status of all cilium pods
+// and returns an error if any of the pods report an unhealthy status
+func (kub *Kubectl) CiliumStatusPreFlightCheck() error {
+	err := kub.ciliumStatusPreFlightCheck()
+	status := "OK"
+	if err != nil {
+		status = err.Error()
+	}
+	ginkgoext.By("Checking that cilium status is healthy: %s", status)
+	return err
+}
+
+func (kub *Kubectl) ciliumControllersPreFlightCheck() error {
 	var controllersFilter = `{range .controllers[*]}{.name}{"="}{.status.consecutive-failure-count}{"\n"}{end}`
 	ciliumPods, err := kub.GetCiliumPods(KubeSystemNamespace)
 	if err != nil {
-		return err
+		return nil
```

Comment:
What is the need for not returning the error if one has occurred?

Reply:
It's a bug

Reply:
@tgraf ping - this would have been helpful in debugging another recent CI failure. Can you please answer my question?

Reply:
It was a bug in my change. Fixed